src/niamoto/core/components/transforms/shapes.py

```
@@ -724,50 +726,86 @@ def _simplify_with_utm(
             gdf = self.load_shape_geometry(geometry_location)
             if gdf is None or gdf.empty:
                 return {}
 
+            # Simplify geometry using UTM projection
             simplified = self._simplify_with_utm(gdf.geometry.iloc[0])
-            return mapping(simplified)
+            return self._convert_to_topojson(simplified)
 
         except Exception as e:
-            raise ValueError(f"Error simplifying geometry: {e}")
+            raise ValueError(f"Error processing geometry: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)